src/main/java/org/javarosa/core/model/instance/utils/InstanceUtils.java

```diff
@@ -0,0 +1,41 @@
+package org.javarosa.core.model.instance.utils;
+
+import static org.javarosa.core.model.instance.utils.TreeUtilities.xmlToTreeElement;
+
+import org.javarosa.core.model.instance.AbstractTreeElement;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.model.instance.InstanceBase;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.xml.util.InvalidStructureException;
+
+import java.io.IOException;
+
+/**
+ * Collection of static instance loading methods
+ *
+ * @author Phillip Mates
+ */
+public class InstanceUtils {
+
+    public static FormInstance loadFormInstance(String formFilepath)
+            throws InvalidStructureException, IOException {
+        TreeElement root = xmlToTreeElement(formFilepath);
+        return new FormInstance(root, null);
+    }
+
+    /**
+     * Sets instance properties to the given instance root
+     *
+     * @param instanceRoot instance root
+     * @param instanceId   instance id to set
+     * @param instanceBase instance base to set
+     */
+    public static void setUpInstanceRoot(AbstractTreeElement instanceRoot, String instanceId,
```

Comment:
should this complain if it's not a `TreeElement`?

Reply:
https://github.com/dimagi/commcare-core/pull/1215/commits/a95afe09f18477fa1b911dfbf2d9781602681ed0